exec/totemip.c

```diff
@@ -267,6 +268,53 @@ const char *totemip_print(const struct totem_ip_address *addr)
 	return (inet_ntop(addr->family, addr->addr, buf, sizeof(buf)));
 }
 
+static unsigned char default_bindnet_addr[sizeof(struct in6_addr)] = {0};
+static struct ifaddrs *totemip_getif_scopeid_ifap = NULL;
+
+static void totemip_getif_scopeid_ifap_reset(void)
+{
+	if (totemip_getif_scopeid_ifap != NULL) {
+		freeifaddrs(totemip_getif_scopeid_ifap);
+	}
+	totemip_getif_scopeid_ifap = NULL;
+}
+
+static int totemip_getif_scopeid(const unsigned char* addr16, unsigned int *scopeid)
+{
+	struct ifaddrs *ifa;
+	const struct sockaddr_in6 *sin;
+	const socklen_t addr_len = sizeof(struct in6_addr);
+	int rc = -1; // 0 = found 1 match; -1 = found 0 matches; -2 = found >1 matches
+	int is_refresh = (NULL == totemip_getif_scopeid_ifap);
+
+	if (is_refresh && getifaddrs(&totemip_getif_scopeid_ifap) != 0)
+		return (-1);
+
+	for (ifa = totemip_getif_scopeid_ifap; ifa; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL || ifa->ifa_netmask == NULL)
+			continue ;
+
+		if ((ifa->ifa_addr->sa_family != AF_INET6) ||
+		    (ifa->ifa_netmask->sa_family != AF_INET6 &&
+		     ifa->ifa_netmask->sa_family != 0))
+			continue ;
+
+		sin = (const struct sockaddr_in6 *)ifa->ifa_addr;
+
+		if (memcmp(&sin->sin6_addr, addr16, addr_len) == 0) {
+			*scopeid = sin->sin6_scope_id;
+			rc = (-1 == rc) ? 0 : -2;
+			if (-2 == rc && is_refresh) {
```

Comment:
It will print the message for every next repeating IP. If we want it print only once, I could bring it out of the loop.